From dbadc258867c14f80f98c8427b8383ae0cc82ac1 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Fri, 12 Oct 2018 17:52:45 +0300
Subject: [PATCH 4857/5242] MLK-18789: arch: arm64: fsl-imx8qxp: Move
 conflicting pin

commit  9630c660e16bf3a6322eb3489ae3d7a70a2ba273 from
https://source.codeaurora.org/external/imx/linux-imx.git

The SC_P_MCLK_OUT0_ADMA_ACM_MCLK_OUT0 pin was conflicting with the pins
needed by the eLCDIF, so it was commented out with a TODO.
In order to fix this conflict, re-write this pinctrl group in the lcdif
specific dts file so that the conflict is handled properly.

Fixes: 471c6dc358fe ("ARM64: dts: imx8dx: Add dts file for lcdif")
Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts   |   10 ++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts  |    2 --
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts
index 4ac48e8..e884a77 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-lcdif.dts
@@ -52,6 +52,16 @@
 
 };
 
+&iomuxc {
+	imx8qxp-mek {
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+				SC_P_COMP_CTL_GPIO_1V8_3V3_GPIORHB_PAD	0x000514a0
+			>;
+		};
+	};
+};
+
 &esai0 {
 	status = "disabled";
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index dc7ee27..51399a7 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -264,9 +264,7 @@
 	imx8qxp-mek {
 		pinctrl_hog: hoggrp {
 			fsl,pins = <
-				/* TODO: conflicts with LCDIF!!!
 				SC_P_MCLK_OUT0_ADMA_ACM_MCLK_OUT0	0x0600004c
-				 */
 				SC_P_COMP_CTL_GPIO_1V8_3V3_GPIORHB_PAD	0x000514a0
 			>;
 		};
-- 
1.7.9.5

